python/rikai/spark/types/video.py

```diff
@@ -93,3 +94,41 @@ def deserialize(self, datum) -> "YouTubeVideo":
 
     def simpleString(self) -> str:
         return "YouTubeVideoType"
+
+
+class SegmentType(UserDefinedType):
+    """Defines the Spark UDT for a video segment"""
+
+    def __init__(self):
+        super().__init__()
+
+    def __repr__(self) -> str:
+        return "SegmentType"
+
+    @classmethod
+    def sqlType(cls) -> StructType:
+        return StructType(
+            fields=[
+                StructField("start_fno", IntegerType(), nullable=False),
```

Comment:
It might be just my ignorance. what is `fno`? mind to add some doc?

Reply:
frame number. I'm using fno to keep consistent with opencv terminology (the assumption is that, whether right or wrong, most ML people working with videos in python will use opencv)